### PR TITLE
Role cert with derived state and its new config

### PR DESCRIPTION
**Pull request by mlajkim**
# Description
Refactor RoleCert feature with DerivedState.
Related Issue: https://github.com/AthenZ/k8s-athenz-sia/issues/113

## Assignees
- [x] `Assignees` is set

## Type of changes
- [x] Apply one or more `labels` of the following that fits:
  - `bug`: Bug fix
  - `dependencies`: Dependency upgrades
  - `documentation`: Documentation changes
  - `enhancement`: New Feature
  - `good first issue`: First contribution
  - `logging`: Log changes
  - `refactor`: Refactoring (no functional changes, no api changes)

### Flags
```
- [ ] Breaks backward compatibility
- [ ] Requires a documentation update
- [ ] Has untestable code
```
---

## Checklist
```
- [ ] Followed the guidelines in the CONTRIBUTING document
- [ ] Added prefix `[skip ci]`/`[ci skip]`/`[no ci]`/`[skip actions]`/`[actions skip]` in the PR title if necessary
- [ ] Tested and linted the code
- [ ] Commented the code
- [ ] Made corresponding changes to the documentation
```

## Checklist for maintainer
```
- [ ] Use `Squash and merge`
- [ ] Double-confirm the merge message has prefix `[skip ci]`/`[ci skip]`/`[no ci]`/`[skip actions]`/`[actions skip]`
- [ ] Delete the branch after merge
```

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -195,10 +200,10 @@ func (idCfg *IdentityConfig) loadFromFlag(program string, args []string) error {
...
-	f.StringVar(&idCfg.RoleCertDir, "rolecert-dir", idCfg.RoleCertDir, "directory to write role certificate files (required for role certificate provisioning)")
-	// RoleCertFilenameDelimiter
-	f.BoolVar(&idCfg.RoleCertKeyFileOutput, "rolecert-key-file-output", idCfg.RoleCertKeyFileOutput, "output role certificate key file (true/false)")
+	f.StringVar(&idCfg.rawTargetDomainRoles, "target-domain-roles", idCfg.rawTargetDomainRoles, "target Athenz roles with domain (e.g. athenz.subdomain"+idCfg.roleCertFilenameDelimiter+"admin,sys.auth"+idCfg.roleCertFilenameDelimiter+"providers) (required for role certificate and token provisioning)")
+	f.StringVar(&idCfg.roleCertDir, "rolecert-dir", idCfg.roleCertDir, "directory to write role certificate files (required for role certificate provisioning)")
+	// roleCertFilenameDelimiter
+	f.BoolVar(&idCfg.roleCertKeyFileOutput, "rolecert-key-file-output", idCfg.roleCertKeyFileOutput, "output role certificate key file (true/false)")
```
Comment (mlajkim) [inline pkg/config/config.go:206]: warning

Comment (mlajkim) [reply pkg/config/config.go:206]: `idCfg.roleCertFilenameDelimiter`
`idCfg.RoleCert.Delimiter`

File: pkg/config/derived-access-token.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
+// Copyright 2023 LY Corporation
```
Comment (y-myajima) [inline pkg/config/derived-access-token.go:1]: derived-access-token.goだけ先にファイルを作成している意図はありますか？
先に作るのであれば、role-tokenも作成しておいた方が良い気がします

File: pkg/config/derived-role-cert.go [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+// Package config defines all the configuration parameters. It reads configuration from environment variables and command-line arguments.
+package config
+
+import (
+	"github.com/AthenZ/k8s-athenz-sia/v3/third_party/log"
+)
+
+type DerivedRoleCert struct {
+	Use               bool         // if fetching role certificate is enabled (de facto standard)
+	Dir               string       // directory to store role certificates. Usually one, but can be multiple // TODO: make it string[]
```
Comment (y-myajima) [inline pkg/config/derived-role-cert.go:24]: `Usually one, but can be multiple // TODO: make it string[]`
こちらのコメントの意図はなんでしょうか？(Naming Formatを使うとDirがなくなると思っていいます)

File: pkg/config/derived-access-token.go (hunk 2 above)
Comment (mlajkim) [reply pkg/config/derived-access-token.go:1]: just a sample to help next requester to follow the format. that is it.

File: pkg/config/derived-role-cert.go (hunk 3 above)
Comment (mlajkim) [reply pkg/config/derived-role-cert.go:24]: mb. old comments left over. will fix it

Comment (mlajkim) [reply pkg/config/derived-role-cert.go]: handled in https://github.com/AthenZ/k8s-athenz-sia/pull/145/commits/7acec25ad6663a912ff6ae8e22f6d6110fae9e12

**Review by y-myajima — APPROVED** [review]
LGTM

**Pull request merged**